### ISSUE TITLE
How to add custom attributes to shape object?

### ISSUE THREAD

**Issue by Sabih23** [scanny/python-pptx#582]
I am looking for a way to add an attribute to shape object which will have another shape object as its value. Lets say I have a shape object, then I want to have an attribute named child_shape:
`shape.child_shape = another_shape_object`

I have tried adding this attribute in the init of base.py class:
`    def __init__(self, shape_elm, parent):
        super(BaseShape, self).__init__()
        self._element = shape_elm
        self._parent = parent
        self.child_shape  = None`

I set its value to None initially but then after some work I assign it the desired value.
`shape.child_shape = another_shape`

Doing it this way only sets the attribute value once, but then if I print the child_shape attribute again from the:
`shape = prs.slides[0].shapes[0]
print(shape.child_shape)`

It always returns None.

Is there a way to make it write this attribute to the xml of the pptx?

**Comment by scanny** [comment]
No. The XML vocabulary is pre-determined by the XMLSchema for PresentationML and the implementation of that vocabulary (at least the subset we use) is by the classes in the `pptx.oxml` subpackage. The "proxy" classes just call the oxml layer to read and write XML. So simply adding an attribute at the proxy layer doesn't have any affect on the XML.
